stress/shell/src/main/java/alluxio/stress/cli/Benchmark.java

```
@@ -0,0 +1,131 @@
+    }
+  }
+
+  /**
+   * Runs the test and returns the string output.
+   *
+   * @param args the command-line args
+   * @return the string result output
+   */
+  public String run(String[] args) throws Exception {
+    JCommander jc = new JCommander(this);
+    jc.setProgramName(this.getClass().getSimpleName());
+    try {
+      jc.parse(args);
+    } catch (Exception e) {
+      jc.usage();
+      throw e;
+    }
+
+    // prepare the benchmark.
+    prepare();
+
+    AlluxioConfiguration conf = new InstancedConfiguration(ConfigurationUtils.defaults());
+    String className = this.getClass().getCanonicalName();
+
+    if (mBaseParameters.mCluster) {
+      // run on job service
+
+      // remove the cluster flag
+      List<String> commandArgs =
+          Arrays.stream(args).filter((s) -> !BaseParameters.CLUSTER_FLAG.equals(s))
+              .filter((s) -> !s.isEmpty()).collect(Collectors.toList());
+
+      commandArgs.addAll(mBaseParameters.mJavaOpts);
+
+      JobInfo jobInfo =
+          JobGrpcClientUtils.run(new StressBenchConfig(className, commandArgs), 0, conf);
+      return jobInfo.getResult().toString();
+    }
+
+    // run locally
+    if (mBaseParameters.mInProcess) {
+      // run in process
+      T result = runLocal();
+      if (mBaseParameters.mDistributed) {
+        return result.toJson();
+      }
+
+      // aggregate the results
+      return result.aggregator().aggregate(Collections.singletonList(result)).toJson();
+    } else {
+      // Spawn a new process
+      List<String> command = new ArrayList<>();
+      command.add(conf.get(PropertyKey.HOME) + "/bin/alluxio");
+      command.add("runClass");
+      command.add(className);
+      command.addAll(Arrays.asList(args));
+      command.add(BaseParameters.IN_PROCESS_FLAG);
+      command.addAll(mBaseParameters.mJavaOpts);
+
+      LOG.info("running command: " + String.join(" ", command));
+      return ShellUtils.execCommand(command.toArray(new String[0]));
+    }
+  }
```

should we throw an exception if neither `mInProcess` or `mCluster` is true?